primitives/core/src/traits.rs

```diff
@@ -139,24 +146,195 @@ pub trait BareCryptoStore: Send + Sync {
 	/// sign the provided message with that key.
 	///
 	/// Returns a tuple of the used key and the SCALE encoded signature.
-	fn sign_with_any(
+	async fn sign_with_any(
 		&self,
 		id: KeyTypeId,
 		keys: Vec<CryptoTypePublicPair>,
 		msg: &[u8]
 	) -> Result<(CryptoTypePublicPair, Vec<u8>), Error> {
 		if keys.len() == 1 {
-			return self.sign_with(id, &keys[0], msg).map(|s| (keys[0].clone(), s));
+			return self.sign_with(id, &keys[0], msg).await.map(|s| (keys[0].clone(), s));
 		} else {
-			for k in self.supported_keys(id, keys)? {
-				if let Ok(sign) = self.sign_with(id, &k, msg) {
+			for k in self.supported_keys(id, keys).await? {
+				if let Ok(sign) = self.sign_with(id, &k, msg).await {
 					return Ok((k, sign));
 				}
 			}
 		}
 		Err(Error::KeyNotSupported(id))
 	}
 
+	/// Sign with all keys
+	///
+	/// Provided a list of public keys, sign a message with
+	/// each key given that the key is supported.
+	///
+	/// Returns a list of `Result`s each representing the SCALE encoded
+	/// signature of each key or a Error for non-supported keys.
+	async fn sign_with_all(
+		&self,
+		id: KeyTypeId,
+		keys: Vec<CryptoTypePublicPair>,
+		msg: &[u8],
+	) -> Result<Vec<Result<Vec<u8>, Error>>, ()> {
+		let futs = keys.iter()
+			.map(|k| self.sign_with(id, k, msg));
+
+		Ok(join_all(futs).await)
+	}
+
+	/// Generate VRF signature for given transcript data.
+	///
+	/// Receives KeyTypeId and Public key to be able to map
+	/// them to a private key that exists in the keystore which
+	/// is, in turn, used for signing the provided transcript.
+	///
+	/// Returns a result containing the signature data.
+	/// Namely, VRFOutput and VRFProof which are returned
+	/// inside the `VRFSignature` container struct.
+	///
+	/// This function will return an error in the cases where
+	/// the public key and key type provided do not match a private
+	/// key in the keystore. Or, in the context of remote signing
+	/// an error could be a network one.
+	async fn sr25519_vrf_sign(
+		&self,
+		key_type: KeyTypeId,
+		public: &sr25519::Public,
+		transcript_data: VRFTranscriptData,
+	) -> Result<VRFSignature, Error>;
+}
+
+/// [deprecated] Sync version of the CryptoStore
```

Comment:
The runtime will never support async crypto store.

So, I'm not really sure that this is deprecated.